pkg/acs/calacs/acsrej/acsrej_init.c

```diff
@@ -27,266 +27,156 @@
-                } else {
-                    Pix(sg->err.data,i,j) = (nse[1]) / exp2n;
-                }
-            } /* End loop over SECOND AMP used on pixels in the line */
+            } /* End loop over ALL AMPS used on pixels in the line */
         } /* End loop over lines */
 
     /* use the minimum to construct the initial average */
     } else {
         if (strncmp(par->initgues, "minimum", 3) != 0) {
-            sprintf (MsgText,"Invalid INITGUES value %s, reset it to 'minimum'", par->initgues);
+            sprintf (MsgText,
+                     "Invalid INITGUES value %s, reset it to 'minimum'",
+                     par->initgues);
             trlwarn (MsgText);
             strcpy (par->initgues, "minimum");
         }
 
         for (n = 0; n < nimgs; n++) {
             initHdr(&dqhdr);
-            getHeader(ipdq[n],&dqhdr);
-            for (j = 0; j < dim_y; j++) {
-                /* Set up the gain and noise values used for this line in ALL images */
-                if (j < ampy ) {
-                    gn[0] = gain2[AMP_C];
-                    gn[1] = gain2[AMP_D];
-                    nse[0] = noise2[AMP_C];
-                    nse[1] = noise2[AMP_D];
-                } else {
-                    gn[0] = gain2[AMP_A];
-                    gn[1] = gain2[AMP_B];
-                    nse[0] = noise2[AMP_A];
-                    nse[1] = noise2[AMP_B];
-                }
+            getHeader(ipdq[n], &dqhdr);
 
-                getFloatLine (ipsci[n], j, buf);
+            for (j = 0; j < dim_y; j++) {
+                getFloatLine (ipsci[n], j, buf);  /* electrons */
                 getShortLine (ipdq[n], j, bufdq);
 
-                /* AMPS C and D */
-                for (i = 0; i < ampx; i++) {
-
-                    raw = buf[i] / gn[0];
-                    raw0 = (raw > 0.) ? raw : 0.;
-                    signal0 = ((raw - skyval[n] / gn[0]) > 0.) ? (raw - skyval[n] / gn[0]) : 0.;
+                /* ALL AMPS */
+                for (i = 0; i < dim_x; i++) {
+                    raw = buf[i];  /* e */
+                    dumf = raw - skyval[n];  /* e */
 
                     if (efac[n] > 0.) {
-                        val = (raw - skyval[n] / gn[0]) / efac[n];
+                        /* Can be negative */
+                        val = dumf / efac[n];  /* e/s */
                     } else {
                         val = 0.;
                     }
 
-                    if ( (n == 0) || (val < Pix(sg->sci.data,i,j)) ) {
+                    if ( (n == 0) || (val < Pix(sg->sci.data, i, j)) ) {
+                        /* If this pixel is bad in the first image,
+                           then the min is automatically set to 0. As a
+                           result, only negative val is going to be stored and
+                           valid positive min is ignored.
+                           SLIGHTLY BUGGY HERE??? */
```

Comment:
Note to self: Agreed this algorithm is buggy, and it pre-dates any changes made by PLL.   Algorithm requires image with exposure time > 0 AND good pixel value in order to set a viable minimum.  Re-think algorithm.  Determine default method: median or minimum.